packages/web-app-files/tests/unit/views/LocationPicker.spec.js

```diff
@@ -0,0 +1,461 @@
+import LocationPicker from '@files/src/views/LocationPicker.vue'
+import { mount, RouterLinkStub, shallowMount } from '@vue/test-utils'
+
+const component = { ...LocationPicker, mounted: jest.fn() }
+
+const defaultStubs = {
+  'oc-breadcrumb': true,
+  'oc-grid': true,
+  'oc-button': true,
+  'oc-icon': true,
+  'oc-table-files': true
+}
+
+localVue.prototype.$client.files = {
+  list: async () => [],
+  move: async () => {}
+}
+localVue.prototype.$client.publicFiles = {
+  list: async () => [],
+  move: async () => {}
+}
+
+const router = {
+  push: jest.fn()
+}
+
+const listLoaderStub = 'list-loader-stub'
+const breadcrumbStub = 'oc-breadcrumb-stub'
+const listInfoStub = 'list-info-stub'
+const filesPaginationStub = 'pagination-stub'
+const translateStub = 'translate-stub'
+
+const selectors = {
+  selectionInfo: '.location-picker-selection-info',
+  currentHint: '[data-testid="location-picker-current-hint"]',
+  cancelButton: '#location-picker-btn-cancel',
+  confirmButton: '#location-picker-btn-confirm',
+  filesView: '#files-view',
+  noContentMessage: '#files-location-picker-empty',
+  filesTable: '#files-location-picker-table',
+  paginationCurrentItem: '.oc-pagination-list-item-current'
+}
+
+describe('LocationPicker', () => {
+  // the function fires up wile route is changed and quickly sets loading to true
+  // so needs to be mocked to get a loading wrapper
+  jest.spyOn(LocationPicker.methods, 'navigateToTarget').mockImplementation()
+
+  describe('files app bar', () => {
+    it.each([
+      { action: 'move', item: '' },
+      { action: 'copy', item: '' },
+      { action: 'move', item: '/parent/child' },
+      { action: 'move', item: '/parent' },
+      { action: 'copy', item: '/parent/child' },
+      { action: 'copy', item: '/parent' },
+      { action: 'random-action', item: '/सिम्पल फोल्डर/टेश्ट' },
+      { action: '', item: '/parent/child' },
+      { action: true, item: '/parent/child' },
+      { action: {}, item: '/parent/child' },
+      {
+        action: 'random-action',
+        item: '/सिम्पल फोल्डर/टेश्ट फोल्डर'
+      },
+      {
+        action: 'random-action',
+        item: '/123/456'
+      },
+      {
+        action: 'random-action',
+        item: '/😂/🔥'
+      }
+    ])('should show location picker selection info according to route action and item', input => {
+      const wrapper = getShallowWrapper({
+        $route: getRoute({ action: input.action, item: input.item })
+      })
+
+      const selection = wrapper.find(selectors.selectionInfo)
+
+      expect(selection.exists()).toBeTruthy()
+      expect(selection).toMatchSnapshot()
+    })
+    it.each(['copy', 'move'])(
+      'should show current hint according to the current route action',
+      action => {
+        const wrapper = getShallowWrapper({ $route: getRoute({ action: action }) })
+        const currentHint = wrapper.find(selectors.currentHint)
+
+        expect(currentHint.exists()).toBeTruthy()
```

Comment:
Checking for the currentHint's existence here and in the snapshot test below is redundant, I'd vote in favor of removing the data-test-id and rely on the snapshot (which should have the `<p>` tag included)

Reply:
Relying on snapshots can easily lead to false negatives... see reasoning for it here https://github.com/owncloud/web/pull/5697#discussion_r710183617

Reply:
current hint takes content from computed property `currentHint` according to the `currentAction` (i.e.move or other). So checking for existence and the snapshot has different intentions.